src/main/java/com/epam/reportportal/extension/azure/AzureExtension.java

```diff
@@ -184,4 +224,165 @@ private void removeListeners() {
+		try {
+            List<WorkItemTypeFieldWithReferences> issueTypeFields = issueTypeFieldsApi
+                    .workItemTypesFieldList(organizationName, projectName, issueType, API_VERSION, EXPAND);
+
+            for (WorkItemTypeFieldWithReferences field : issueTypeFields) {
+				WorkItemField detailedField = null;
+                // Skip fields that return 404 on request
+				try {
+					detailedField = fieldsApi
+							.fieldsGet(organizationName, field.getReferenceName(), projectName, API_VERSION);
+				} catch (ApiException e) {
+					continue;
+				}
+
+                // Skip read-only fields
+				if (detailedField.isReadOnly()) {
+					continue;
+				}
+
+                List<AllowedValue> allowed = new ArrayList<>();
+				switch (detailedField.getName()) {
+                    // Skip Work Item Type field cause we have the same custom field
+					case "Work Item Type":
+						continue;
+					case "Iteration ID":
+						for (WorkItemClassificationNode node : iterationNodes) {
+							allowed.add(new AllowedValue(node.getId().toString(), node.getName()));
+						}
+						break;
+					case "Area ID":
+						for (WorkItemClassificationNode node : areaNodes) {
+							allowed.add(new AllowedValue(node.getId().toString(), node.getName()));
+						}
+						break;
+					default:
+						for (Object value : field.getAllowedValues()) {
+							allowed.add(new AllowedValue(value.toString().replace(" ", "_"), value.toString()));
+						}
+						break;
+				}
+
+                // Add an empty line to each field with a non-empty list of allowed values
+				if (allowed.size() > 0) {
+                    allowed.add(0, new AllowedValue("Empty_String", ""));
+                }
+
+				List<String> defaultValue = new ArrayList<>();
+				if (field.getDefaultValue() != null) {
+					defaultValue.add(field.getDefaultValue().toString().replace(" ", "_"));
+				}
+
+                ticketFields.add(new PostFormField(field.getReferenceName().replace(".", "_"), field.getName(),
+                        detailedField.getType().toString(), field.isAlwaysRequired(), defaultValue, allowed));
+			}
+
+            List<PostFormField> sortedTicketFields = ticketFields.stream()
+                    .sorted(Comparator.comparing(PostFormField::getIsRequired).reversed()
+                            .thenComparing(PostFormField::getFieldName)).collect(Collectors.toList());
+
+			// Add to the top a custom field representing the work item type
+			sortedTicketFields.add(0, new PostFormField("issuetype", "Issue Type", "issuetype", true,
+                    List.of(issueType), new ArrayList<AllowedValue>()));
+
+			return sortedTicketFields;
+		} catch (ApiException e) {
+            LOGGER.error("Unable to load ticket fields: " + e.getMessage(), e);
+			throw new ReportPortalException(ErrorType.UNABLE_INTERACT_WITH_INTEGRATION,
+					"Unable to load ticket fields. Code: " + e.getCode() + ", Message: " + e.getMessage(), e);
```

Comment:
Better way is using something like  String.format("Hello! My name is %s, I'm %s.", name, age) instead of straight string concatenation. it's more readable. Use it all the time in any place. Find out the best way in Java. Java is big.

Reply:
Fixed